src/main.js

```diff
@@ -336,19 +336,35 @@ class MiniGraphCard extends LitElement {
       </div>` : '';
   }
 
+  generateLegend(entity, index) {
+    let legend = this.computeName(index);
+    const state = this.getEntityState(index);
+    const show_legend_state = this.config.entities[index];
```

Comment:
Still not getting this.
You gave a bool `show_legend_state` option inside "entities" object.
Why not using it?
Instead you use `this.config.entities[index];` - which is an object, not a bool.

Reply:
I didn't inspect what was really going on in this object
It should be better now